app/src/main/java/com/github/se/eduverse/model/folder/FolderViewModel.kt

```diff
@@ -0,0 +1,95 @@
+package com.github.se.eduverse.model.folder
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+class FolderViewModel(val repository: FolderRepository) : ViewModel() {
+
+  private val _existingFolders: MutableStateFlow<MutableList<Folder>> =
+      MutableStateFlow(
+          repository
+              .getFolders(
+                  {},
+                  { Log.e("FolderViewModel", "Exception $it while trying to load the folders") })
+              .toMutableList())
+  val existingFolders: StateFlow<MutableList<Folder>> = _existingFolders
+
+  var activeFolder: MutableStateFlow<Folder?> = MutableStateFlow(null)
```

Comment:
I think that this should be a private val _activeFolder and offered as a StateFlow like done for existingFolders

Reply:
Done